tests/Integration/BaseUserMetaEndpointTest.php

```diff
@@ -145,4 +145,22 @@ protected function send_put_request( string $data ): string {
 
 		return $result;
 	}
+
+	/**
+	 * Verifies that the route is not registered when the respective filter is
+	 * set to false.
+	 */
+	public function run_test_do_not_register_route_when_proxy_is_disabled(): void {
+		// Override some setup steps in order to set the filter to false.
+		remove_action( 'rest_api_init', $this->rest_api_init_proxy );
+		$endpoint                  = $this->get_endpoint();
+		$this->rest_api_init_proxy = static function () use ( $endpoint ) {
+			add_filter( 'wp_parsely_enable_' . self::$filter_key . '_endpoint', '__return_false' );
```

Comment:
I think that making the aforementioned filter change would also result in a change here.

```suggestion
			add_filter( 'wp_parsely_enable_' . self::$filter_key . '_api_proxy', '__return_false' );
```